src/components/PopupWithForm.js

```diff
@@ -1,11 +1,21 @@
 import Popup from "./Popup.js";
 
 class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleFormSubmit, buttonText, loadingButtonText) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
-    this._button = this._popupElement.querySelector(".modal__button");
+    this._button = this._popupElement.querySelector(".modal__button-submit");
     this._handleFormSubmit = handleFormSubmit;
+    this._buttonText = buttonText;
+    this._loadingButtonText = loadingButtonText;
+  }
+
+  showLoading() {
+    this._button.textContent = this._loadingButtonText;
+  }
+
+  hideLoading() {
+    this._button.textContent = this._buttonText;
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```